### PR TITLE
Gsijb/gxl helpers

### PR DESCRIPTION
**Pull request by jtbenson116**
merging FVS integration to main

### CODE REVIEW

**Review by sourcesync — CHANGES_REQUESTED** [review]
@jtbenson116  Hello! Can you please add jupyterlab and matplotlib (with preferred versions) to the requirements.txt? Thanks!

**Review by sourcesync — CHANGES_REQUESTED** [review]
@jtbenson116 thanks for the recent changes.  Can you please add an rsync script that populates the data files into the "results/" directory from nas1 or nas2?

**Review by sourcesync — CHANGES_REQUESTED** [review]
@jtbenson116 thanks for the changes - i can run the notebook now with the sync'ed data.  Can you check out the error in the cell "In [20]" - I pushed the [notebook in this branch](https://github.com/jtbenson116/vector-db-benchmark/blob/gsijb/gxl-helpers/results.ipynb) so you can see the error...

**Review by sourcesync — CHANGES_REQUESTED** [review]
@jtbenson116 thanks!  OK new set of requests:
- can you add the deep-1B data files to the rsync as well?  Maybe rsync to a local "./data" directory that gets excluded via .gitignore
- it looks like you need to add the weaviate data files also to the rsync script.  Currently, the notebook expected to find them under /home/jacob

**Review by sourcesync — CHANGES_REQUESTED** [review]
@jtbenson116 OK finally getting to this PR.  If I enable only the qdrant tests in run_all_engines.sh and run it, I get the following below.  Is it correct?

```
(qdrant-benchmarking-jacob-fork) gwilliams@sv7-apu11:~/Projects/FVS/qdrant_vdbb_fork_from_jacob/vector-db-benchmark$ ./run_all_engines.sh 
The authenticity of host 'localhost (::1)' can't be established.
ECDSA key fingerprint is SHA256:Iubk2A/C1O0xVXAM9XXDvuKI/fHlt+MYJqzu4m22gGk.
Are you sure you want to continue connecting (yes/no/[fingerprint])? y
Please type 'yes', 'no' or the fingerprint: yes
Warning: Permanently added 'localhost' (ECDSA) to the list of known hosts.
qdrant@localhost's password: 
```

**Comment by jtbenson116** [comment]
I never use `run_all_engines.sh` so I'm not sure.

**Comment by sourcesync** [comment]
ok @jtbenson116 thanks.  Can you refresh my memory - what script should I use to, for example, just run the Qdrant benchmarks first?

**Comment by jtbenson116** [comment]
`run.py --engine <engine config> --datasets <dataset name>`

File: experiments/configurations/gsi.json [hunk 1]
```diff
@@ -0,0 +1,586 @@
...
+        "name": "gsi-768-clusters-noam1",
+        "engine": "gsi",
+        "upload_params": {"nbits": 768, "searchType": "clusters", "top": 10, "nprobe": 200, "nprobe_rerank": 100, "hammingk": 250},
+        "search_params": [{"top": 10}]
+    },
+    {
+        "name": "gsi-128-clusters-noam2",
+        "engine": "gsi",
+        "upload_params": {"nbits": 128, "searchType": "clusters", "top": 10, "nprobe": 200, "nprobe_rerank": 100, "hammingk": 250,  "num_of_boards": 1, "num_of_clusters": 128000},
+        "search_params": [{"top"}: 10}]
```
Comment (sourcesync) [inline experiments/configurations/gsi.json]: @jtbenson116 I was getting a JSON parser error, I think there is an extra end brace here.

File: experiments/configurations/gsi.json [hunk 2]
```diff
@@ -0,0 +1,586 @@
...
+    },
+    {
+        "name": "gsi-256-clusters-noam2",
+        "engine": "gsi",
+        "upload_params": {"nbits": 256, "searchType": "clusters", "top": 10, "nprobe": 200, "nprobe_rerank": 100, "hammingk": 250,  "num_of_boards": 1, "num_of_clusters": 128000},
+        "search_params": [{"top"}: 10}]
```
Comment (sourcesync) [inline experiments/configurations/gsi.json]: @jtbenson116 I was getting a JSON parser error, I think there is an extra end brace here.

File: experiments/configurations/gsi.json [hunk 3]
```diff
@@ -0,0 +1,586 @@
...
+    },
+    {
+        "name": "gsi-512-clusters-noam2",
+        "engine": "gsi",
+        "upload_params": {"nbits": 512, "searchType": "clusters", "top": 10, "nprobe": 200, "nprobe_rerank": 100, "hammingk": 250,  "num_of_boards": 1, "num_of_clusters": 128000},
+        "search_params": [{"top"}: 10}]
```
Comment (sourcesync) [inline experiments/configurations/gsi.json]: @jtbenson116 I was getting a JSON parser error, I think there is an extra end brace here.

File: experiments/configurations/gsi.json [hunk 4]
```diff
@@ -0,0 +1,586 @@
...
+    },
+    {
+        "name": "gsi-768-clusters-noam2",
+        "engine": "gsi",
+        "upload_params": {"nbits": 768, "searchType": "clusters", "top": 10, "nprobe": 200, "nprobe_rerank": 100, "hammingk": 250,  "num_of_boards": 1, "num_of_clusters": 128000},
+        "search_params": [{"top"}: 10}]
```
Comment (sourcesync) [inline experiments/configurations/gsi.json]: @jtbenson116 I was getting a JSON parser error, I think there is an extra end brace here.

**Comment by sourcesync** [comment]
@jtbenson116 OK maybe you can help me out.  I think I started the qdrant docker containers correctly and running this:

```
python3 run.py --engines qdrant-m-16-ef-128 --datasets deep-image-96-angular
```

I get this error:

```
Experiment stage: Upload
9990000it [06:08, 27086.18it/s]
Upload time: 368.9612747039646
Total import time: 597.3474990827963
Experiment stage: Search
10000it [01:02, 160.11it/s]
10000it [01:20, 123.73it/s]
10000it [01:46, 93.87it/s]
10000it [03:19, 50.18it/s]
10000it [00:51, 195.68it/s]
10000it [01:12, 138.21it/s]
10000it [01:51, 90.08it/s]
10000it [03:16, 51.01it/s]
10000it [00:47, 210.54it/s]
10000it [01:08, 146.89it/s]
10000it [01:48, 92.55it/s]
10000it [03:03, 54.46it/s]
10000it [00:36, 272.86it/s]
10000it [01:04, 154.04it/s]
Experiment qdrant-m-16-ef-128 - deep-image-96-angular interrupted
Traceback (most recent call last):
  File "/home/gwilliams/Projects/Qdrant_benchmarking/jacob-fork/vector-db-benchmark/run.py", line 54, in run
    client.run_experiment(dataset, skip_upload, skip_search)
  File "/home/gwilliams/Projects/Qdrant_benchmarking/jacob-fork/vector-db-benchmark/engine/base_client/client.py", line 86, in run_experiment
    search_stats = searcher.search_all(
  File "/home/gwilliams/Projects/Qdrant_benchmarking/jacob-fork/vector-db-benchmark/engine/base_client/search.py", line 108, in search_all
    zip(*pool.imap_unordered(search_one, iterable=tqdm.tqdm(queries)))
  File "/home/gwilliams/anaconda3/envs/qdrant-benchmarking-jacob-fork/lib/python3.10/multiprocessing/pool.py", line 873, in next
    raise value
multiprocessing.pool.MaybeEncodingError: Error sending result: '<multiprocessing.pool.ExceptionWithTraceback object at 0x7fa8b3f0a0e0>'. Reason: 'TypeError("cannot pickle '_thread.RLock' object")'
Traceback (most recent call last):

  File "/home/gwilliams/Projects/Qdrant_benchmarking/jacob-fork/vector-db-benchmark/run.py", line 83, in <module>
    app()

  File "/home/gwilliams/Projects/Qdrant_benchmarking/jacob-fork/vector-db-benchmark/run.py", line 76, in run
    raise e

  File "/home/gwilliams/Projects/Qdrant_benchmarking/jacob-fork/vector-db-benchmark/run.py", line 54, in run
    client.run_experiment(dataset, skip_upload, skip_search)

  File "/home/gwilliams/Projects/Qdrant_benchmarking/jacob-fork/vector-db-benchmark/engine/base_client/client.py", line 86, in run_experiment
    search_stats = searcher.search_all(

  File "/home/gwilliams/Projects/Qdrant_benchmarking/jacob-fork/vector-db-benchmark/engine/base_client/search.py", line 108, in search_all
    zip(*pool.imap_unordered(search_one, iterable=tqdm.tqdm(queries)))

  File "/home/gwilliams/anaconda3/envs/qdrant-benchmarking-jacob-fork/lib/python3.10/multiprocessing/pool.py", line 873, in next
    raise value

multiprocessing.pool.MaybeEncodingError: Error sending result: '<multiprocessing.pool.ExceptionWithTraceback object at 0x7fa8b3f0a0e0>'. Reason: 'TypeError("cannot pickle '_thread.RLock' object")'
```

**Comment by jtbenson116** [comment]
Oh yeah. As far as I can tell there is no bug here. I've seen the same error when I ran Qdrant benchmarks, it comes from the very end of the tests though (over all engine configs)  so there isn't any issue with the benchmark results.

File: engine/base_client/search.py [hunk 5]
```diff
@@ -5,9 +5,11 @@
 
 import numpy as np
 import tqdm
+import os
 
 from dataset_reader.base_reader import Query
 
+
 DEFAULT_TOP = 10
```
Comment (sourcesync) [inline engine/base_client/search.py:13]: @jtbenson116 Does this mean K=10 for TopK?

**Review by sourcesync — CHANGES_REQUESTED** [review]
@jtbenson116 Can you address my feedback in this review?

**Pull request merged**